File: eNote/eNote.Model/DTOs/Kurs.cs
using eNote.Model.Enums;

namespace eNote.Model.DTOs
{
    public class Kurs
    {
        public int Id { get; set; }
        public string Naziv { get; set; } = null!;
        public string? Opis { get; set; }
        public decimal Cijena { get; set; }
        public int BrojUpisanih { get; set; }
        public string? DatumPocetka { get; set; }
        public string? DatumZavrsetka { get; set; }
        public string InstruktorIme { get; set; } = null!;
    }
}

File: eNote/UI/enote_desktop/lib/models/kurs.dart
import 'package:json_annotation/json_annotation.dart';

part 'kurs.g.dart';

@JsonSerializable()
class Kurs {
  int? id;
  String? naziv;
  String? opis;
  double? cijena;
  int? brojUpisanih;
  String? datumPocetka;
  String? datumZavrsetka;
  int? instruktorId;
  String? instruktorIme;

  Kurs();

  factory Kurs.fromJson(Map<String, dynamic> json) => _$KursFromJson(json);

  Map<String, dynamic> toJson() => _$KursToJson(this);
}
